File: chapter 2/coffee-price.pl
use strict;
use warnings;

use LWP::Simple;

my $price = 99.9;
while ($price > 4.74) {
    sleep(900);
    my $page = get("http://beans.itcarlow.ie/prices.html");
    die "Couldn't get it!" unless defined $page;
    
    my $index = index($page, '>$') + 2;
    my $last = 4;
    
    my $price = substr($page, $index, $last);
}

print "Buy!\n";

File: chapter 2/coffee-price.py
import urllib.request
import time

price = 99.9
while price > 4.74:
    time.sleep(900)
    page = urllib.request.urlopen("http://beans.itcarlow.ie/prices.html")
    text = page.read().decode("utf8")
    
    index = text.find(">$") + 2
    last = index + 4
    
    price = float(text[index:last])

print("Buy!")